app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -315,63 +315,36 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
         }
     }
 
-    fun getRecentChanges(
-        driveId: Int,
-        onlyFirstPage: Boolean,
-        forceDownload: Boolean = false
-    ): LiveData<FileListFragment.FolderFilesResult?> {
+    fun getRecentChanges(driveId: Int, onlyFirstPage: Boolean, forceDownload: Boolean = false): LiveData<FolderFilesResult?> {
         getRecentChangesJob.cancel()
         getRecentChangesJob = Job()
-
         val ignoreDownload = lastModifiedTime != 0L && (Date().time - lastModifiedTime) < DOWNLOAD_INTERVAL && !forceDownload
-
         return liveData(Dispatchers.IO + getRecentChangesJob) {
             if (ignoreDownload) {
-                emit(
-                    FileListFragment.FolderFilesResult(
-                        files = FileController.getRecentChanges(),
-                        isComplete = true,
-                        page = 1
-                    )
-                )
-                return@liveData
+                emit(FolderFilesResult(files = FileController.getRecentChanges(), isComplete = true, page = 1))
+            } else {
+                getLastModifiedFilesRecursively(1, driveId, onlyFirstPage)
             }
+        }
+    }
 
-            suspend fun recursive(page: Int) {
-                val isFirstPage = page == 1
-                val apiResponse = ApiRepository.getLastModifiedFiles(driveId, page)
-                if (apiResponse.isSuccess()) {
-                    val data = apiResponse.data
-                    data?.let { FileController.storeRecentChanges(it, isFirstPage) }
-                    when {
-                        data == null -> Unit
-                        data.size < ApiRepository.PER_PAGE -> {
-                            if (isFirstPage) emit(
-                                FileListFragment.FolderFilesResult(files = data, isComplete = true, page = page)
-                            )
-                        }
-                        else -> {
-                            if (isFirstPage) {
-                                emit(
-                                    FileListFragment.FolderFilesResult(files = data, isComplete = true, page = page)
-                                )
-                            }
-                            if (!onlyFirstPage) recursive(page + 1)
-                        }
-                    }
-                } else {
-                    if (isFirstPage) {
-                        emit(
-                            FileListFragment.FolderFilesResult(
-                                files = FileController.getRecentChanges(),
-                                isComplete = true,
-                                page = 1
-                            )
-                        )
-                    } else emit(null)
+    private suspend fun LiveDataScope<FolderFilesResult?>.getLastModifiedFilesRecursively(
```

Comment:
Since you are moving it, I think keeping the same name with the word `recursive` at the end would be more coherent. Because it belongs to him anyway.

Example : `getRecentChangesRecursive`